packages/uxpin-merge-cli/src/steps/serialization/component/implementation/typescript/component/getComponentName.ts

```diff
@@ -1,8 +1,14 @@
 import { TSSerializationContext } from '../context/getSerializationContext';
 import { getComponentFileName } from './getComponentFileName';
 import { ComponentDeclaration } from './getPropsTypeAndDefaultProps';
+import * as ts from 'typescript'
 
 export function getComponentName(context:TSSerializationContext, component:ComponentDeclaration):string {
-  return component.name && component.name.getText()
-    || getComponentFileName(context);
+  if(component.name) {
+    return component.name.getText()
+  }
+  if (ts.isArrowFunction(component) && component.parent && (component.parent as ts.ArrowFunction).name){
```

Comment:
A comment on this case would be great

```suggestion
  // Support named arrow functions and use the component parent's name.
  // https://github.com/UXPin/uxpin-merge-tools/issues/208
  if (ts.isArrowFunction(component) && component.parent && "name" in component.parent && (component.parent as ts.ArrowFunction).name) {
```

I also think that check could be slightly safer so I added that too